drivers/accel/adxl314/adxl314.c

```diff
@@ -0,0 +1,1146 @@
+		/* If data rate smaller than 12.5 Hz set to this val. */
+		else if (dev->odr < ADXL314_ODR_12_5HZ)
+			ret = adxl314_set_odr(dev, ADXL314_ODR_12_5HZ);
+		if (ret)
+			return ret;
+
+		dev->lp_mode = ADXL314_LP_MODE_ON;
+	} else {
+		dev->lp_mode = ADXL314_LP_MODE_NONE;
+	}
+
+	/* Clear the AUTOSLEEP bit. */
+	ret = adxl314_control_reg_bit(dev, ADXL314_REG_BW_RATE,
+				      enable, ADXL314_LOW_POWER_OP);
+	if (ret)
+		return ret;
+
+	/* Set MEAS operation mode. */
+	ret = adxl314_set_op_mode(dev, ADXL314_MEAS);
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Activate/Deactivate AUTOSLEEP.
+ *
+ * @param dev        - The device structure.
+ * @param enable     - Setting: 0 - disable autosleep;
+ *                           1 - enable autosleep.
+ * @param inact_thr  - Inactivity threshold value.
+ * @param time_inact - Time inactivity value.
+ *
+ * @return ret       - Result of the setting operation action.
+ *                     0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int adxl314_set_autosleep_mode(struct adxl314_dev *dev, enum bit_action enable,
+			       uint8_t inact_thr, uint8_t time_inact)
+{
+	int ret = -EFAULT;
+	uint8_t mask = ADXL314_POWER_CTL_LINK | ADXL314_POWER_CTL_AUTO_SLEEP;
+	uint8_t data = ADXL314_POWER_CTL_LINK | ADXL314_POWER_CTL_AUTO_SLEEP;
+
+	/* Set STDBY operation mode. */
+	ret = adxl314_set_op_mode(dev, ADXL314_STDBY);
+	if (ret)
+		return ret;
+
+	if (enable == ENABLE_E) {
+		ret = adxl314_conf_inact_thr(dev, inact_thr);
+		if (ret)
+			return ret;
+
+		ret = adxl314_conf_time_inact(dev, time_inact);
+		if (ret)
+			return ret;
+
+		dev->lp_mode = ADXL314_LP_MODE_AUTOSLEEP;
+	} else {
+		ret = adxl314_conf_inact_thr(dev, 0);
+		if (ret)
+			return ret;
+
+		ret = adxl314_conf_time_inact(dev, 0);
+		if (ret)
+			return ret;
+
+		data = mask & (~data);
+
+		dev->lp_mode = ADXL314_LP_MODE_NONE;
+	}
+
+	/* Set the LINK and AUTOSLEEP bits. */
+	ret = adxl314_reg_write_msk(dev, ADXL314_REG_POWER_CTL, data, mask);
+	if (ret)
+		return ret;
+
+	/* Set MEAS operation mode. */
+	ret = adxl314_set_op_mode(dev, ADXL314_MEAS);
+
+	return ret;
+}
+
+/*******************************************************************************
+ * @brief Configure interrupt enable control.
+ *
+ * @param dev         - The device structure.
+ * @param en_ctl      - Enable control configuration.
+ *
+ * @return ret        - Result of the configuration action.
+ *                      0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int adxl314_conf_int_enable(struct adxl314_dev *dev,
+			    union adxl314_int_en_reg_flags en_ctl)
+{
+	int ret = -EINVAL;
+	uint8_t reg_val = en_ctl.value;
+
+	ret = adxl314_write_device_data(dev, ADXL314_REG_INT_EN, 1, &reg_val);
```

Comment:
ditto: return directly. Please check all functions for which this statement applies